### PR TITLE
Improve error messages for `Union` types in `Credentials`

### PR DESCRIPTION
**Pull request by shreyamalviya**
Related to https://github.com/guardicore/monkey/issues/3942

~Test script to see errors raised, with outputs: [error_messages_tests.zip](https://github.com/guardicode/monkeytypes/files/13715489/error_messages_tests.zip)~ --> Just look at the UTs.

~(This only works with dictionary inputs right now.)~

### CODE REVIEW

File: monkeytypes/credentials/credentials.py [hunk 1]
```diff
@@ -1,15 +1,65 @@
...
+        return keys[0]
+
+    return None
+
+
+Identity = Annotated[
+    Union[Annotated[Username, Tag("username")], Annotated[EmailAddress, Tag("email_address")]],
+    Discriminator(
+        get_discriminator_value_identity,
+        custom_error_type=INVALID_UNION_MEMBER_ERROR,
```
Comment (mssalvatore) [inline monkeytypes/credentials/credentials.py:43]: Can we use pydantic's `union_tag_not_found` instead of having our own custom one?

Comment (mssalvatore) [reply monkeytypes/credentials/credentials.py:43]: NVM

**Review by ilija-lazoroski — APPROVED** [review]

**Review by mssalvatore — CHANGES_REQUESTED** [review]
I have the following issues when running unit tests:
```
__________________________________________________________ ERROR collecting tests/test_sshkeypair.py __________________________________________________________

ImportError while importing test module '/home/msalvatore/git/monkeytypes/tests/test_sshkeypair.py'.

Hint: make sure your test modules/packages have valid Python names.

Traceback:

/usr/lib/python3.11/importlib/__init__.py:126: in import_module

    return _bootstrap._gcd_import(name[level:], package, level)

tests/test_sshkeypair.py:3: in <module>

    from monkeytypes import SSHKeypair

monkeytypes/__init__.py:44: in <module>

    from .credentials.credentials import Credentials, CredentialsComponent, Identity, Secret

monkeytypes/credentials/__init__.py:11: in <module>

    from .credentials import Credentials, CredentialsComponent, Identity, Secret

monkeytypes/credentials/credentials.py:5: in <module>

    from pydantic import Discriminator, Tag, field_serializer

E   ImportError: cannot import name 'Discriminator' from 'pydantic' (/home/msalvatore/git/monkeytypes/.venv/lib/python3.11/site-packages/pydantic/__init__.py)

=================================================================== short test summary info ===================================================================

ERROR tests/test_agent_plugin_manifest.py

ERROR tests/test_base_models.py

ERROR tests/test_credentials.py

ERROR tests/test_email_address.py

ERROR tests/test_file_extension.py

ERROR tests/test_int_range.py

ERROR tests/test_lm_hash.py

ERROR tests/test_network_range.py

ERROR tests/test_nt_hash.py

ERROR tests/test_password.py

ERROR tests/test_percent.py

ERROR tests/test_socket_address.py

ERROR tests/test_sshkeypair.py

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 13 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

===================================================================== 13 errors in 0.22s ======================================================================

```

If I'm reading the changelog correctly, it looks like you're using v2.5 features but we're using pydantic v2.4.

**Review by mssalvatore — CHANGES_REQUESTED** [review]
flake8 check fails. Are your pre-commit hooks working?

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**